provider/bitmovinnewsdk/types/custom_data.go

```diff
@@ -0,0 +1,27 @@
+package types
+
+import "errors"
+
+// CustomData holds data to set on encodings and configurations to hold relationship state
+type CustomData *map[string]map[string]interface{}
```

Comment:
Oh just noticed this. Does it need to be a pointer?

```suggestion
type CustomData map[string]map[string]interface{}
```

Or could we have some kind of structured data on our side that then gets marshaled into a JSON-object that looks arbitrary in the bitmovin API? Something like:

```go
type CustomData struct {
  Audio struct {
    ID string `json:"id"`
  } `json:"audio"`
  Container struct {
    Name string `json:"name"`
  } `json:"container"`
  Manifest struct {
    ID string `json:"id"`
  } `json:"manifest"`
}
```

(not necessarily with inline structs, I'm just being lazy here 🙈 )